resources/api/api-spec.yml

```diff
@@ -121,6 +121,11 @@ components:
       items:
         type: string
 
+    keyName:
```

Comment:
Is it necessary to add validation rules here?

My only concern is that it is too easy for these rules to become out of sync with the actual API validations.

Reply:
Horrifically we decided to validate the query server side and to rely on exceptions. 

Additionally the pattern is not enforced by the code generator and no verification is done by the client because considered as out of scope.